base/profile.jl

```diff
@@ -640,14 +640,14 @@ function rtruncto(str::String, w::Int)
     if length(str) <= w
         return str
     else
-        return string("...", str[end-w+4:end])
+        return string("...", str[chr2ind(str, length(str)-w+4):end])
```

Comment:
It likely won't matter in practice but it might be better to use `endof(str)` to ensure that it's a valid index.

Reply:
Unfortunately no as `endof(str)` returns byte index.
What we want to do here is go back `w-4` **characters** from `endof(str)`. We could run `w+4` times `prevind` from `endof(str)`, but this is cumbersome so that is why I have used `chr2ind` and performed the calculations on character indices not byte indices and only converted them to byte index at the end with `chr2ind` (which is inefficient, but I guess in this application it should not be a problem).
I hope to make such calculations simpler and more efficient with #23765 in the future.